### PR TITLE
Create Home View

### PR DESCRIPTION
**Pull request by minoring**
Create Home View with navigation drawer and bottom navigation bar. #5

### CODE REVIEW

**Comment by minoring** [comment]
@mtnviewlab Immediately committed .gitignore and removed from PR. Thank you

File: Frontend/app/src/androidTest/java/io/happybugs/happybugs/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package io.happybugs.happybugs;
+
+import android.content.Context;
```
Comment (shberry) [inline Frontend/app/src/androidTest/java/io/happybugs/happybugs/ExampleInstrumentedTest.java]: Group all imports in one group, and all static imports in another group. https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing

File: Frontend/app/src/androidTest/java/io/happybugs/happybugs/ExampleInstrumentedTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
+    @Test
+    public void useAppContext() {
+        // Context of the app under test.
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
```
Comment (shberry) [inline Frontend/app/src/androidTest/java/io/happybugs/happybugs/ExampleInstrumentedTest.java]: Remove extra line?

File: Frontend/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
+        <activity android:name=".MainActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
Comment (shberry) [inline Frontend/app/src/main/AndroidManifest.xml]: Remove extra line?

File: Frontend/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+    </application>
+
```
Comment (shberry) [inline Frontend/app/src/main/AndroidManifest.xml]: Extra line

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.TextView;
+
+import io.happybugs.happybugs.R;
+
+public class MainActivity extends AppCompatActivity
+        implements NavigationView.OnNavigationItemSelectedListener {
+
+    private TextView mTextMessage;
```
Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: Can we come up with a better variable name?

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+import android.support.v4.view.GravityCompat;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.TextView;
+
+import io.happybugs.happybugs.R;
```
Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: Can we come up with a better name than R?

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                    mTextMessage.setText(R.string.title_home);
+                    return true;
+                case R.id.navigation_create:
+                    mTextMessage.setText(R.string.title_create);
+                    return true;
+                case R.id.navigation_account:
+                    mTextMessage.setText(R.string.title_account);
+                    return true;
+            }
+            return false;
```
Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: Another option is to add a default case in switch and add 'return false' there. Up to you.

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        if (drawer.isDrawerOpen(GravityCompat.START)) {
+            drawer.closeDrawer(GravityCompat.START);
+        } else {
+            super.onBackPressed();
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
```
Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: If we are never returning false, why are we even returning a boolean value here?

Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: Do you know if there's a case when we need to return false?

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        return super.onOptionsItemSelected(item);
+    }
+
+    @SuppressWarnings("StatementWithEmptyBody")
+    @Override
+    public boolean onNavigationItemSelected(MenuItem item) {
+        // Handle navigation view item clicks here.
+        int id = item.getItemId();
+
+        if (id == R.id.nav_camera) {
```
Comment (shberry) [inline Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: Can you add // TODO(yourname): blah blah ?

File: Frontend/app/src/main/res/values/styles.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<resources>
+
```
Comment (shberry) [inline Frontend/app/src/main/res/values/styles.xml]: Remove extra lines. I see too many extra lines here and there :)

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java (hunk 5 above)
Comment (minoring) [reply Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: It's a testing message whether navigation bar works properly.
I think it's not proper to have test message in member variable, so I'll remove it.

File: Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java (hunk 6 above)
Comment (minoring) [reply Frontend/app/src/main/java/io/happybugs/happybugs/MainActivity.java]: I think R is generated android studio by default.

**Comment by mtnviewlab** [comment]
I decline merge. Please recreate PR.

**Pull request closed without merging**